### PR TITLE
DB-2094 add nscdr configs for logstash

### PR DESCRIPTION
**Pull request by sincejune**

### CODE REVIEW

File: log-collection/ansible/roles/logstash/templates/configs/nscdr_1.logstash.conf.j2 [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+filter {
+  mutate {
+    rename => {
+      "orig_callid" => "call-id"
+    }
+  }
+  date {
+    match => ["time_insert", "yyyy-MM-dd HH:mm:ss"]
```
Comment (natalie-arsky) [inline log-collection/ansible/roles/logstash/templates/configs/nscdr_1.logstash.conf.j2:25]: I looked at the sample data provided and time_insert is often has value of NULL. We need to choose another time value that is always present to be event timestamp. May be time_release?  It's unix timestamp, example: time_release:1589880754.
Can we check in the data we have that it's always present?

File: log-collection/ansible/roles/logstash/tasks/main.yml [hunk 2]
```diff
@@ -85,16 +85,43 @@
...
+- name: Create Netsapiens CDR cron job to update mysql statement based on current date
   cron:
     name: "logstash restart with new mysql_statement"
     minute: "0"
     hour: "0"
+    day: "1"
     job: |-
-      systemctl stop logstash; rm /etc/logstash/conf.d/.logstash_jdbc_last_run*; echo "SELECT event_index, hostname, event_type, orig_callid, event_text, event_ts, event_time FROM event_$(date +\%Y\%m\%d) WHERE event_type='responder' AND event_index > :sql_last_value">/etc/logstash/conf.d/netsapiens_mysql_statement; systemctl start logstash
-  #when: "'netsapiens' in configs_list"
+      systemctl stop logstash; rm /etc/logstash/conf.d/.logstash_jdbc_last_run_nscdr*; echo "SELECT * FROM $(date +\%Y\%m)_r AND id > :sql_last_value">/etc/logstash/conf.d/nscdr_mysql_statement; systemctl start logstash
```
Comment (natalie-arsky) [inline log-collection/ansible/roles/logstash/tasks/main.yml:126]: Should be
"SELECT * FROM $(date +\%Y\%m)_r WHERE id > :sql_last_value"

File: log-collection/ansible/roles/logstash/templates/configs/nscdr_1.logstash.conf.j2 (hunk 1 above)
Comment (sincejune) [reply log-collection/ansible/roles/logstash/templates/configs/nscdr_1.logstash.conf.j2]: Fixed in https://github.com/dashbase/dashbase-installation/pull/195/commits/a7832fe5c70a7a59a49029a1edcd4e7b1ba4a301

Used `time_start` as the timestamp and confirmed `time_start` is a Non-Null field in table definition.

File: log-collection/ansible/roles/logstash/tasks/main.yml (hunk 2 above)
Comment (sincejune) [reply log-collection/ansible/roles/logstash/tasks/main.yml]: fixed.

**Comment by sincejune** [comment]
I found there's a serious issue we need to work out.

We used [tracking_column](https://www.elastic.co/guide/en/logstash/current/plugins-inputs-jdbc.html#plugins-inputs-jdbc-tracking_column) to record the last run position while it only supports `numeric` and `timestamp` fields([ref](https://www.elastic.co/guide/en/logstash/current/plugins-inputs-jdbc.html#plugins-inputs-jdbc-tracking_column_type)). I looked at the sample data, `id` is a hexadecimal string and it's not in ascending order. So we can not use it in this scenario.

Instead, I used `time_start` as the tracking column with statement `WHERE time_start >= :sql_last_value` in this PR. The disadvantage is that it can produce many duplicated events because `time_start` is not unique. (Maybe we should use `time_release` here, but I'm not so sure if `time_release` can be `NULL`, and, it has the same "duplicated events" issue)

Any thoughts about this? @natalie-arsky

**Comment by natalie-arsky** [comment]
Regarding sql_last_value: I think it's OK to use time_start. There is no difference between using time_start vs time_release, both will end up with duplicated events. I will look at the data we have to see if there is another column we can use but if not, having duplicated events for 1 second every 10 min is not that bad. This may be the best we can do.

**Pull request merged**